config/src/main/java/org/springframework/security/config/annotation/web/configurers/ExpressionUrlAuthorizationConfigurer.java

```diff
@@ -188,6 +190,14 @@ final ExpressionBasedFilterInvocationSecurityMetadataSource createMetadataSource
 			if (trustResolver != null) {
 				defaultHandler.setTrustResolver(trustResolver);
 			}
+			ApplicationContext context = http.getSharedObject(ApplicationContext.class);
+			if(context != null) {
```

Comment:
There is no guarantee the context has been set (it is optional). This feature is only available if we have access to the ApplicationContext.